homeassistant/components/webostv/config_flow.py

```diff
@@ -0,0 +1,203 @@
+            or user_input is not None
+        ):
+            try:
+                client = await async_control_connect(self._host, None)
+            except PyLGTVPairException:
+                return self.async_abort(reason="error_pairing")
+            except WEBOSTV_EXCEPTIONS:
+                errors["base"] = "cannot_connect"
+            else:
+                data = {CONF_HOST: self._host, CONF_CLIENT_SECRET: client.client_key}
+                return self.async_create_entry(title=self._name, data=data)
+
+        return self.async_show_form(step_id="pairing", errors=errors)
+
+    async def async_step_ssdp(self, discovery_info):
+        """Handle a flow initialized by discovery."""
+        self._host = urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname
+        self._name = discovery_info[ssdp.ATTR_UPNP_FRIENDLY_NAME]
+
+        uuid = discovery_info[ssdp.ATTR_UPNP_UDN]
+        if uuid.startswith("uuid:"):
+            uuid = uuid[5:]
+        await self.async_set_unique_id(uuid)
+        self._abort_if_unique_id_configured({CONF_HOST: self._host})
+
+        for progress in self._async_in_progress():
+            if progress.get("context", {}).get(CONF_HOST) == self._host:
+                return self.async_abort(reason="already_in_progress")
+
+        self._uuid = uuid
+        return await self.async_step_pairing()
+
+
+class OptionsFlowHandler(config_entries.OptionsFlow):
+    """Handle options."""
+
+    def __init__(self, config_entry):
+        """Initialize options flow."""
+        self.config_entry = config_entry
+        self.options = config_entry.options
+        self.host = config_entry.data[CONF_HOST]
+        self.key = config_entry.data[CONF_CLIENT_SECRET]
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        errors = {}
+        if user_input is not None:
+            on_action = user_input.get(CONF_ON_ACTION)
+
+            if on_action:
+                action_state = self.hass.states.get(on_action)
+
+                if not action_state or action_state.domain != "script":
+                    errors["base"] = "script_not_found"
+
+            if not errors:
+                options_input = {
+                    CONF_ON_ACTION: on_action,
+                    CONF_SOURCES: user_input[CONF_SOURCES],
+                }
+                return self.async_create_entry(title="", data=options_input)
+
+        # Get turn on service
+        script_turn_on = self.options.get(CONF_ON_ACTION, "")
+
+        # Get sources
+        sources = self.options.get(CONF_SOURCES, "")
+        sources_list = await async_default_sources(self.host, self.key)
+        if sources_list is None:
+            errors["base"] = "cannot_retrieve"
+
+        options_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_ON_ACTION,
+                    description={"suggested_value": script_turn_on},
```

Comment:
Do we have an existing integration that offers this option in an options flow? We may need some more eyes on this to make sure we're going down the right path for this kind of option.

Reply:
I inherited it from the original PR #50352, my initial thought was to try to implement wake on lan so that an external script will not be needed, but after some thinking I realized there may be many different ways to turn on the TV device, since this device is not listening to the network when off (except for wake on lan on models it is supported) it gets complicated, people may use HDMI-CEC, a smart plug to toggle the power or sending IR command. As it is now it is the least breaking option from what currently implemented, although I agree not ideal.

Reply:
In the philips_js integration we just support device triggers to support this feature.